h2o-docs/src/product/data-science/upliftdrf.rst

```diff
@@ -0,0 +1,454 @@
+
+Uplift DRF can be applied in fields where we operate with two groups of subjects. Fist group, let's call it treatment, receive some kind of treatment (marketing campaign, medicine,...), and a second group, let's call it control, is separated from the treatment. We also gather information about their response, whether they bought a product, recover from disease, and similar. Then, Uplift DRF trains so-called uplift trees. Uplift trees take information about treatment/control group assignment and information about response directly into a decision about splitting a node. The output of the uplift model is the probability of change in user behavior which helps to decide if treatment impacts the desired behavior (buy a product, recover from disease,...). In other words, if a user responds because the user was treated. This leads to proper campaign targeting on a subject that genuinely needs to be treated and avoids wasting resources on subjects that respond/do not respond anyway.
+
+The current version of Uplift DRF is based on implementation of DRF. The principle of training is similar to DRF. When given a set of data, Uplift DRF generates a forest of classification uplift trees, rather than a single classification tree. Each of these trees is a weak learner built on a subset of rows and columns. More trees will reduce the variance. Classification take the average prediction over all of their trees to make a final prediction. (Note: For a categorical response column, Uplift DRF maps factors  (e.g. 'dog', 'cat', 'mouse) in lexicographic order to a name lookup array with integer indices (e.g. 'cat -> 0, 'dog' -> 1, 'mouse' -> 2.)
+
+
+Uplift metrics
+~~~~~~~~~~~~~~
+
+Uplift DRF differentiates itself from DRF because it finds the best split using ``treatment_column``. The goal is split training customers into a group which get an offer (treatment group) and a group which does not (control group). This information (``treatment_column``) with features and ``response_column`` are used for training. The ``uplift_metric`` is calculated to decide which point from the histogram is selected to split the data in the tree node (instead of calculation squared error like in other tree algorithms).
+
+The goal is to maximize the differences between the class distributions in the treatment and control sets, so the splitting criteria are based on distribution divergences. The ditribution divergence is calculated based on the ``uplift_metric`` parameter. In H2O-3, three ``uplift_metric`` types are supported:
+
+- **Kullback-Leibler divergence** (``uplift_metric="kl"``) - uses logarithms to calculate divergence, asymmetric widely used, tends to infinity values (if treatment or control group distributions contain zero values). :math:`KL(P, Q) = \sum_{{i=0}^{N} p_i \log{\frac{p_i}{q_i}} }`
```

Comment:
There is missing comma i suppose. 
" asymmetric widely used"
 asymmetric, widely used

Reply:
fixed